[GStreamer] Audio is distorted in Kaltura videos

philn opened the pull request:
#### b87a52fd5788fcd275446bb61c2ca49b8f85430a
<pre>
[GStreamer] Audio is distorted in Kaltura videos
<a href="https://bugs.webkit.org/show_bug.cgi?id=297116">https://bugs.webkit.org/show_bug.cgi?id=297116</a>

Reviewed by Xabier Rodriguez-Calvar.

Re-allow decodebin(3) to autoplug avdec_aac decoders after making sure the MSE AAC parser emits
frames with ADTS headers. Raw AAC doesn&apos;t seem very well supported in avdec_aac, leading to garbled
audio rendering.

The fdkaacdec decoder handles raw AAC better, but depends on FDK-AAC which might not be shipped by
every distro, or requires explicit installation from non-free repositories.

Canonical link: <a href="https://commits.webkit.org/299318@main">https://commits.webkit.org/299318@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/884dd5bf77e95ee804d2e2828017ff9a72b16e2f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118222 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37900 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28539 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124381 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70263 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120100 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38593 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46482 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89710 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59348 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121175 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30740 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105997 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70206 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29810 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24116 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68047 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100166 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24290 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127455 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45125 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34017 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98392 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45486 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102218 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98179 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25023 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43576 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21564 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41590 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44997 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50671 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44457 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47802 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46146 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented:
Keeping as draft because this patch uncovered a likely bug in aacparse which outputs buffers without duration and un-contiguous timestamps when configured to generate ADTS headers :/

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5650248cdd710eb5985307efd937ce038eae29e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49814

philn committed:
[GStreamer] Audio is distorted in Kaltura videos
https://bugs.webkit.org/show_bug.cgi?id=297116

Reviewed by Xabier Rodriguez-Calvar.

Re-allow decodebin(3) to autoplug avdec_aac decoders after making sure the MSE AAC parser emits
frames with ADTS headers. Raw AAC doesn't seem very well supported in avdec_aac, leading to garbled
audio rendering.

The fdkaacdec decoder handles raw AAC better, but depends on FDK-AAC which might not be shipped by
every distro, or requires explicit installation from non-free repositories.

Canonical link: https://commits.webkit.org/299318@main

webkit-commit-queue commented:
Committed 299318@main (b87a52fd5788): <https://commits.webkit.org/299318@main>

Reviewed commits have been landed. Closing PR #49814 and removing active labels.
